torchaudio/datasets/cmudict.py

```diff
@@ -0,0 +1,92 @@
+from collections import OrderedDict
+import os
+import re
+from pathlib import Path
+from typing import Tuple, Union, List, Dict
+
+from torch.utils.data import Dataset
+from torchaudio.datasets.utils import download_url
+
+_CHECKSUMS = {
+    "http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b":
+    "825f4ebd9183f2417df9f067a9cabe86",
+    "http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b.symbols":
+    "385e490aabc71b48e772118e3d02923e",
+}
+
+
+class CMUDICT(Dataset):
+    """Create a Dataset for CMU Pronouncing Dictionary (CMUDict).
+
+    Args:
+        root (str or Path): Path to the directory where the dataset is found or downloaded.
+        url (str, optional):
+            The URL to download the dictionary from.
+            (default: ``"http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b"``)
+        url_symbols (str, optional):
+            The URL to download the list of symbols from.
+            (default: ``"http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b.symbols"``)
+        download (bool, optional):
+            Whether to download the dataset if it is not found at root path. (default: ``False``).
+    """
+
+    def __init__(self,
+                 root: Union[str, Path],
+                 url: str = "http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b",
+                 url_symbols: str = "http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b.symbols",
+                 download: bool = False) -> None:
+
+        # Get string representation of 'root' in case Path object is passed
+        root = os.fspath(root)
+        if not os.path.isdir(root):
+            os.mkdir(root)
+
+        if download:
+            if not os.path.isdir(root):
```

Comment:
Not sure the logic here is correct.

In L41 and L42, `root` directory is created if it is not a directory, and in L45 it will perform download if `root` is not a directory.

Reply:
Thanks for pointing it out. I've fixed the logics [here](https://github.com/pytorch/audio/pull/1627/files#diff-50ee900e0897ac7528dccd9bdd36d5f0c7cb2d45310aefb46a3bf281f73d88c2R99-R111). It now creates a directory if root is not a directory and only download the data when the directory is there.